Assets/Scripts/UI/Canvas/StateMachine/UIStateManager.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using Framework.PhoneCamera;
+using UnityEngine;
+
+using UI.Canvas.Introduction;
+using UI.Canvas.PhotoBookSystem;
+using UI.Canvas.PhotoTaking;
+
+namespace UI.Canvas.StateMachine
+{
+    public sealed class UIStateManager : MonoBehaviour
+    {
+        [Header("References")]
+        [SerializeField] private IntroductionPopup introductionPopup;
+        [SerializeField] private PhotoBook photoBook;
+        [SerializeField] private GameObject cameraPanel;
+        [SerializeField] private GameObject takingPicture;
+        
+        [Header("Current UI State")]
+        [SerializeField] private UIState currentState;
+
+        /// <summary>
+        /// This switch case checks whether or not you're in the correct state,
+        /// and whether or not you're allowed to move from a certain state to another one.
+        /// </summary>
+        /// <param name="newState">This is the state that is passed in.</param>
+        /// <exception cref="ArgumentOutOfRangeException">This error will be called when the state is unable</exception>
```

Comment:
"when the state is ubable"

Reply:
> "when the state is ubable"

Er staat unable, moet unusable zijn

Reply:
Update het naar unusable in volgende commit.